fiftyone/brain/internal/core/leaky_splits.py

```diff
@@ -1,108 +1,140 @@
 
-from fiftyone import ViewField as F
 import fiftyone.core.brain as fob
-import fiftyone.brain.similarity as sim
-import fiftyone.core.utils as fou
 import fiftyone.core.validation as fov
+import fiftyone.zoo as foz
+from fiftyone import ViewField as F
+
+import fiftyone.brain as fb
+import fiftyone.brain.similarity as fbs
+import fiftyone.brain.internal.core.utils as fbu
+
 
-fbu = fou.lazy_import("fiftyone.brain.internal.core.utils")
+logger = logging.getLogger(__name__)
 
 _DEFAULT_MODEL = "resnet18-imagenet-torch"
+_DEFAULT_BATCH_SIZE = None
 
 
 def compute_leaky_splits(
     samples,
-    brain_key=None,
-    split_views=None,
-    split_field=None,
     split_tags=None,
-    threshold=0.2,
-    similarity_brain_key=None,
-    embeddings_field=None,
+    split_field=None,
+    split_views=None,
+    threshold=None,
+    embeddings=None,
+    similarity_index=None,
     model=None,
     model_kwargs=None,
-    similarity_backend=None,
-    similarity_config_dict=None,
-    **kwargs,
+    batch_size=None,
+    num_workers=None,
+    skip_failures=True,
+    progress=None,
 ):
     """See ``fiftyone/brain/__init__.py``."""
 
-    fov.validate_collection(samples)
+    fov.validate_image_collection(samples)
+
+    if etau.is_str(embeddings):
+        embeddings_field, embeddings_exist = fbu.parse_embeddings_field(
+            samples, embeddings
+        )
+        embeddings = None
+    else:
+        embeddings_field = None
+        embeddings_exist = None
+
+    if etau.is_str(similarity_index):
+        similarity_index = samples.load_brain_results(similarity_index)
+
+    if (
+        model is None
+        and embeddings is None
+        and similarity_index is None
+        and not embeddings_exist
+    ):
+        model = foz.load_zoo_model(_DEFAULT_MODEL)
+        if batch_size is None:
+            batch_size = _DEFAULT_BATCH_SIZE
 
     config = LeakySplitsConfig(
-        split_views,
-        split_field,
-        split_tags,
-        similarity_brain_key,
-        embeddings_field,
-        model,
-        model_kwargs,
-        similarity_backend,
-        similarity_config_dict,
-        **kwargs,
+        split_field=split_field,
+        split_tags=split_tags,
+        embeddings_field=embeddings_field,
+        similarity_index=similarity_index,
+        model=model,
+        model_kwargs=model_kwargs,
     )
+
     brain_method = config.build()
     brain_method.ensure_requirements()
 
-    if brain_key is not None:
-        # Don't allow overwriting an existing run with same key, since we
-        # need the existing run in order to perform workflows like
-        # automatically cleaning up the backend's index
-        brain_method.register_run(samples, brain_key, overwrite=False)
-
-    results = brain_method.initialize(samples, brain_key)
-
-    if results._save_similarity_index:
-        results._similarity_method.register_run(
+    if similarity_index is None:
+        similarity_index = fb.compute_similarity(
```

Comment:
Cleaner to call `compute_similarity()` to generate a similarity index rather than directly interacting the underlying classes as was done in the previous implementation